### ISSUE TITLE
Default Chroma lexer does not support linenos

### ISSUE THREAD

**Issue by ynadji** [gohugoio/hugo#8824]
### What version of Hugo are you using (`hugo version`)?

<pre>
¡ hugo version
hugo v0.86.0+extended linux/amd64 BuildDate=unknown
</pre>

### Does this issue reproduce with the latest release?
v0.86.0 appears to be the latest release, so yes.

### Issue
Here's a small snippet of Zeek (FKA Bro) code that I'd like syntax highlighted with line numbers:

```
{{< highlight bro "linenos=table,linenostart=1" >}}
hook Notice::policy(n: Notice::Info)
    {
    if ( n$note == NewKnownHost )
        add n$actions[Notice::ACTION_TELEGRAM];
    }
```

From the image below, we can see that the syntax highlighting is correct, but there are no line numbers. 

<img width="497" alt="Screen Shot 2021-07-29 at 10 52 04 AM" src="https://user-images.githubusercontent.com/6740/127514953-5be4da41-8a81-4d0e-8d23-fa43c95c51cd.png">

If I change the language to C, it works just fine (albeit with slightly crooked line numbers).

```
{{< highlight c "linenos=table,linenostart=1" >}}
hook Notice::policy(n: Notice::Info)
    {
    if ( n$note == NewKnownHost )
        add n$actions[Notice::ACTION_TELEGRAM];
    }
```

<img width="557" alt="Screen Shot 2021-07-29 at 10 52 26 AM" src="https://user-images.githubusercontent.com/6740/127514975-22417691-60d3-410f-b52d-6b51a143bc20.png">

I suspect this is because Chroma doesn't have a Bro/Zeek lexer, so maybe it's falling back to pygments? But I tried to find another language only in pygments and not in Chroma and couldn't reproduce the error. Would love some direction here so I could try to fix the issue.

**Comment by bep** [comment]
>so maybe it's falling back to pygments?

No, it's falling back to "a default lexer"; I'm a little surprised that that does not support line numbers, though.

**Comment by pamubay** [comment]
Is this related with code fences markdown attributes doesnt work on **chroma-unsupported** language?
Markdown attributes on code fences not parsed if the language is not supported by chroma:

```md
`​`​`bro {.class #id data-attr="val"} 
code
 code
`​`​`
``` 

render:
```html
<pre>
  <code class="language-bro">
    ...
  </code>
</pre>
```

expected: (markdown attributes injected) to `pre`
```html
<pre class="class" id="id" data-attr="val">
  <code class="language-bro">
    ...
  </code>
</pre>
```

**Comment by jmooring** [comment]
The line numbers (@ynadji) and attributes (@pamubay) will be correctly rendered when an unknown language is specified if you enable syntax guessing in the site configuration:

```toml
[markup]
  [markup.highlight]
    guessSyntax = true
```

But, even with this configuration, the line numbers and attributes will be missing if you do not specify a language (or if Hugo can't guess; I'm not sure which).

Related: <https://discourse.gohugo.io/t/guesssyntax/31080>

**Comment by ynadji** [comment]
Adding that to my `config.toml` let's the line numbers render correctly, but eliminates all syntax highlighting.

**Comment by helfper** [comment]
The original issue in this thread is because of this piece of code:
https://github.com/gohugoio/hugo/blob/f27e542442d19436f1428cc22bb03aca398d37a7/markup/highlight/highlight.go#L67-L73

If `lexers.Fallback` is used instead of printing the code block there, the highlight attributes (like line numbers) should work properly.
